### PR TITLE
doc: How to publish N-API package

### PR DESCRIPTION
**Pull request by aruneshchandra**
This guide provides a recommendation on how to publish an experimental N-API package for easy discovery.

### CODE REVIEW

File: locale/en/docs/guides/publishing-napi-modules.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    - Go through the release checklist (ensure tests/demos/docs are OK)
+    - `npm publish --tag n-api`
+
+In this example, tagging the release with `n-api` has ensured that, although 
+version 1.2.0-3 is later than the non-N-API published version (1.2.0-2), it 
+will not be installed if someone chooses to install `iotivity-node` by simply 
+running `npm install iotivity-node`. This will install the non-N-API version 
+by default. The user will have to run `npm install iotivity-node@n-api` to 
+receive the N-API version.
+
```
Comment (lpinca) [inline locale/en/docs/guides/publishing-napi-modules.md]: Nit: what do you think about adding a link to https://docs.npmjs.com/getting-started/using-tags?

**Review by lpinca — APPROVED** [review]

File: locale/en/docs/guides/publishing-napi-modules.md (hunk 1 above)
Comment (aruneshchandra) [reply locale/en/docs/guides/publishing-napi-modules.md]: Done! Thanx

File: locale/en/docs/guides/publishing-napi-modules.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    quite a bit of flexibility to insert any text. e.g. `1.2.0-napi`.
+    - Go through the release checklist (ensure tests/demos/docs are OK)
+    - `npm publish --tag n-api`
+
+In this example, tagging the release with `n-api` has ensured that, although 
+version 1.2.0-3 is later than the non-N-API published version (1.2.0-2), it 
+will not be installed if someone chooses to install `iotivity-node` by simply 
+running `npm install iotivity-node`. This will install the non-N-API version 
+by default. The user will have to run `npm install iotivity-node@n-api` to 
+receive the N-API version. For more information on usig tags with npm check
```
Comment (lpinca) [inline locale/en/docs/guides/publishing-napi-modules.md]: There is a minor typo: "usig".

Comment (aruneshchandra) [reply locale/en/docs/guides/publishing-napi-modules.md]: oops :(
Thanx - fixed it.

**Review by mhdawson — APPROVED** [review]
LGTM

File: locale/en/docs/guides/publishing-napi-modules.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+The following steps are illustrated using the package `iotivity-node`:
+  - First, publish the non-N-API version:
+    - Update the version in `package.json`. For `iotivity-node`, the version 
+    becomes `1.2.0-2`.
+    - Go through the release checklist (ensure tests/demos/docs are OK)
+    - `npm publish`
+  - Then, publish the N-API version:
+    - Update the version in `package.json`. For `iotivity-node`, the version 
+    becomes `1.2.0-3`. For version number, the field after the dash (-) allows 
+    quite a bit of flexibility to insert any text. e.g. `1.2.0-napi`.
```
Comment (refack) [inline locale/en/docs/guides/publishing-napi-modules.md]: [lite suggestion]Maybe link to http://semver.org/

File: locale/en/docs/guides/publishing-napi-modules.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    - Go through the release checklist (ensure tests/demos/docs are OK)
+    - `npm publish --tag n-api`
+
+In this example, tagging the release with `n-api` has ensured that, although 
+version 1.2.0-3 is later than the non-N-API published version (1.2.0-2), it 
+will not be installed if someone chooses to install `iotivity-node` by simply 
+running `npm install iotivity-node`. This will install the non-N-API version 
+by default. The user will have to run `npm install iotivity-node@n-api` to 
+receive the N-API version. For more information on using tags with npm check 
+out https://docs.npmjs.com/getting-started/using-tags
```
Comment (refack) [inline locale/en/docs/guides/publishing-napi-modules.md]: [Suggestion] I would call the link by it's name:
`check out ["Using dist-tags"](https://docs.npmjs.com/getting-started/using-tags).`

File: locale/en/docs/guides/publishing-napi-modules.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+To add the N-API version of `iotivity-node` as a dependency, the `package.json` 
+will look like this:
+
+```JSON
+  "dependencies": {
+    "iotivity-node": "n-api"
+  }
+```
+
+Note that, unlike regular versions, tagged versions cannot be addressed by
```
Comment (refack) [inline locale/en/docs/guides/publishing-napi-modules.md]: [suggestion]
* use new **Note:** convention https://github.com/nodejs/node/pull/13133
* refer to npm guild again
`**Note:** `~that,~` As explained in ["Using dist-tags"](https://docs.npmjs.com/getting-started/using-tags), unlike regular versions`

**Review by refack — APPROVED** [review]
Some lite suggestions

File: locale/en/docs/guides/publishing-napi-modules.md (hunk 5 above)
Comment (gibfahn) [reply locale/en/docs/guides/publishing-napi-modules.md]: And if you're referring to it twice, you might as well use the 

```md
["Using dist-tags"][]


["Using dist-tags"]: https://docs.npmjs.com/getting-started/using-tags
```

format

File: locale/en/docs/guides/publishing-napi-modules.md (hunk 4 above)
Comment (aruneshchandra) [reply locale/en/docs/guides/publishing-napi-modules.md]: Done

File: locale/en/docs/guides/publishing-napi-modules.md (hunk 3 above)
Comment (aruneshchandra) [reply locale/en/docs/guides/publishing-napi-modules.md]: Done

File: locale/en/docs/guides/publishing-napi-modules.md (hunk 5 above)
Comment (aruneshchandra) [reply locale/en/docs/guides/publishing-napi-modules.md]: Done!

**Review by refack — APPROVED** [review]

**Review by gibfahn — APPROVED** [review]

**Comment by aruneshchandra** [comment]
Thanx for all the reviews! Can someone please land this for me ?

**Comment by gibfahn** [comment]
@fhemberger let me know if I did anything wrong.

**Comment by fhemberger** [comment]
@gibfahn No worries, LGTM!

**Comment by aruneshchandra** [comment]
@fhemberger - do we know when this change will show up on the website ? I have a N-API announcement blog coming up next week and would like to point to this content with an official URL.

**Comment by lpinca** [comment]
@aruneshchandra it is visible: https://nodejs.org/en/docs/guides/publishing-napi-modules/. Site is deployed on every push on master.

**Comment by aruneshchandra** [comment]
Ah thanx! I see it now - for some reason I did not see it earlier :/

**Comment by fhemberger** [comment]
@aruneshchandra Site is cached by Cloudflare, it can sometimes take up to an hour for the changes to be visible.

**Pull request merged**